File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/UpdatedUserImpulse.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Impulse which indicates the result of a update user impulse
 */
class UpdatedUserImpulse: Type() {

    fun id() { 
        add {"id" } 
    }

    fun data(config: User.() -> Unit) = 
        add { "data{ " + User().apply(config).render(it) + "}"}

}

File: gaia-sdk-javascript/src/graphql/response/type/UpdatedUserImpulse.ts

import {User} from "./User";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update user impulse
*/
export interface UpdatedUserImpulse {
    id?:Uuid, 
    data?:User
}